supremm/outputter.py

```diff
@@ -21,6 +26,42 @@ def __exit__(self, exception_type, exception_val, trace):
         return self._impl.__exit__(exception_type, exception_val, trace)
 
 
+class FileOutput(object):
+    """
+    Simple outputter that dumps the job summary to a file given by out_file. Intended for debug purposes.
+    """
+    def __init__(self, outconf, resconf):
+        self._resid = resconf['resource_id']
+        self._path = outconf['out_file']
+        if not os.path.exists(self._path):
+            raise Exception("Path specified by out_file does not exist")
+        self._file = open(self._path, 'w')
+        self._dojson = True
+        if self._dojson:
+            self._jsonfile = open(self._path+'.json', 'w')
+            self._jsonarray = []
+
+    def __enter__(self):
+        print ("Calling enter on file", file=self._file)
+        return self
+
+    def process(self, summary, mdata):
+        """
+        json print
+        """ 
+        print(self._resid, json.dumps(summary.get(), indent=4), file=self._file)
+        print("MDATA: ", json.dumps(mdata, indent=4), file=self._file)
+        if self._dojson:
+            self._jsonarray.append(summary.get())
+            self._jsonarray.append(mdata)
+
+    def __exit__(self, exception_type, exception_val, trace):
+        print("Calling exit on {}".format(self._file.name), file=self._file)
+        self._file.close()
```

Comment:
What happens if you call close() on sys.stdout? This was not being done in the previous StdoutOutputter implementation.